### ISSUE TITLE
Is there  a way to execute one job flow with multiple executors?

### ISSUE THREAD

**Issue by samj24** [azkaban/azkaban#1548]
Hello Azkaban Team/@HappyRay,

I have started using Azkaban to schedule the jobs on multi executor environment configuring (i.e. One web server and two executors)
In the job flow one job is dependent on 100+ other independent jobs. please refer to the uploaded image.
Currently, Azkaban executor manager only can submit whole flow to execution server, due to which all 100 jobs are running on single executor server,is there a way to distribute 100 jobs  on multiple executor servers?

In summary:
Is there  a way to execute one job flow with multiple executors?

![image](https://user-images.githubusercontent.com/33263331/32268980-10fdc9d6-bf17-11e7-98fe-4a4e2c20ef8f.png)
Thanks

**Comment by chengren311** [comment]
Hi, thanks for filing the issue. Unfortunately, this is not what we currently support. But it's a potential feature we might consider in the future. Would you mind telling us the concrete gains to distribute the flow to multiple servers in your use case?

**Comment by huangfushun** [comment]
if so I would suggest more lables for executor servers(eg. hostname=ec1, ssd=1) and support more specs for jobs definition like "node affniity"  so schduler manager can choose which server is proper to schdule.

**Comment by samj24** [comment]
@chengren311 ,Thanks for the response.
 
Use Case:
A job flow contains 100+ jobs and each job can consume maximum 15g of memory and 15-40 mins to process. One job among 100+ jobs is dependent on all other jobs and it should start processing once all the other jobs are completed.
 
Problem:
Processing single job flow on a single executor is impacting the scalability and performance as there are many other such job flows need to run.
 
Advantages of distributing a single workflow to multiple servers:
1. Available servers and resources can be utilized efficiently
2. Multiple job flows can run on scale and boost up performance
 
 
Alternate approach:
Send all the jobs in a single job flow excluding the job which is dependent on others and track the completion of sent jobs in the application. Once all the jobs are completed then send the last job which is dependent on other jobs to process. However, UI monitoring would be cumbersome to monitor thousands of jobs using this approach.Thoughts/Alternative?
 
Thanks

**Comment by HappyRay** [comment]
AZ is designed as a lightweight launcher. If you are running many resource-intensive jobs at the same time, is it possible to offload them to an external distributed processing system such as Hadoop/Yarn? e.g. run each job in YARN.
